cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/ApiConnectionInfoBuilder.java

```diff
@@ -48,16 +53,30 @@ public ApiConnectionInfoBuilder withTokenProvider(OAuth2TokenProvider tokenProvi
         return this;
     }
 
-    public ApiConnectionInfoBuilder withCookieJarBuilder(CookieJarFactory.CookieJarBuilder cookieJarBuilder) {
-        this.cookieJarBuilder = cookieJarBuilder;
+    public ApiConnectionInfoBuilder withCookieJarSupplier(CookieJarFactory.CookieJarSupplier cookieJarSupplier) {
+        this.cookieJarSupplier = cookieJarSupplier;
+        return this;
+    }
+
+    public ApiConnectionInfoBuilder withCookieAuthenticator(AuthCookieCallback callback) {
+        cookieAuthenticator = new CookieAuthenticator(callback);
         return this;
     }
 
     public ApiConnectionInfo build() {
         CookieJar cookieJar = null;
-        if (cookieJarBuilder != null) {
-            cookieJar = cookieJarBuilder.buildCookieJar();
+        if (cookieJarSupplier != null) {
+            cookieJar = cookieJarSupplier.getCookieJar();
+        }
+        Authenticator authenticator = null;
+        List<Interceptor> interceptors = new ArrayList<>();
+        if (tokenProvider != null) {
+            authenticator = new OAuth2TokenAuthenticator(tokenProvider);
+            interceptors.add(new OAuth2TokenInterceptor(tokenProvider));
+        }
+        if (cookieAuthenticator != null) {
+            authenticator = cookieAuthenticator;
```

Comment:
isnt this potentially replacing the authenticator you just instantiated?

Reply:
Yes, now throwing an error if you supply both